format/riff/dolby_metadata.go

```diff
@@ -0,0 +1,322 @@
+	d.FieldU8("program_id")
+	// TODO: make struct and read U1(?) U1 (lfeon) U3 (bsmod) U3(acmod) fields?
+	programInfo := d.FieldU8("program_info")
+	lfeon := programInfo & 0b1_000_000
+	bsmod := programInfo & 0b0_111_000
+	acmod := programInfo & 0b0_000_111
+	d.FieldValueBool("lfe_on", lfeon != 0)
+	if bsmod == 0b111 && acmod != 0b001 {
+		bsmod = 0b1000
+	}
+	d.FieldValueStr("bitstream_mode", bitstreamModeMap[bsmod])
+
+	d.FieldU16("ddplus_reserved_a")
+
+	d.FieldU8("surround_config")
+	d.FieldU8("dialnorm_info")
+	d.FieldU8("langcod")
+	d.FieldU8("audio_prod_info")
+	d.FieldU8("ext_bsi1_word1")
+	d.FieldU8("ext_bsi1_word2")
+	d.FieldU8("ext_bsi2_word1")
+
+	d.FieldU24("ddplus_reserved_b")
+
+	d.FieldU8("compr1", scalar.UintSym("reserved"), compressionDescMap)
+	d.FieldU8("dynrng1", scalar.UintSym("reserved"), compressionDescMap)
+
+	d.FieldU24("ddplus_reserved_c")
+
+	d.FieldU8("ddplus_info1")
+
+	d.FieldU40("ddplus_reserved_d")
+
+	d.FieldU16("datarate")
+	d.FieldRawLen("reserved_for_future_use", 69*8)
+}
+
+func parseAudioInfo(d *decode.D) {
+	d.FieldU8("program_id")
+	d.FieldUTF8("audio_origin", 32)
+	d.FieldU32("largest_sample_value")
+	d.FieldU32("largest_sample_value_2")
+	d.FieldU32("largest_true_peak_value")
+	d.FieldU32("largest_true_peak_value_2")
+	d.FieldU32("dialogue_loudness")
+	d.FieldU32("dialogue_loudness_2")
+	d.FieldU32("speech_content")
+	d.FieldU32("speech_content_2")
+	d.FieldUTF8("last_processed_by", 32)
+	d.FieldUTF8("last_operation", 32)
+	d.FieldUTF8("segment_creation_date", 32)
+	d.FieldUTF8("segment_modified_date", 32)
+}
+
+func parseDolbyAtmos(d *decode.D) {
+	d.FieldUTF8NullFixedLen("atmos_dbmd_content_creation_preamble", 32)
+	d.FieldUTF8NullFixedLen("atmos_dbmd_content_creation_tool", 64)
+	d.FieldStruct("version", func(d *decode.D) {
+		d.FieldU8("major")
+		d.FieldU8("minor")
+		d.FieldU8("patch")
+	})
+	// TODO: All these unknowns? (mostly from MediaInfoLib, also Dolby repo)
+
+	d.FieldRawLen("unknown0", 21*8)
+
+	d.FieldRawLen("unknown1", 1)
+	d.FieldU3("downmix_5to2", scalar.UintSym("unknown"), downmix5to2DescMap)
+	d.FieldRawLen("unknown2", 2)
+	d.FieldU2("phaseshift_90deg_5to2", scalar.UintSym("unknown"), phaseShift5to2DescMap)
+
+	d.FieldRawLen("unknown3", 12*8)
+
+	d.FieldRawLen("bed_distribution", 2)
+	d.FieldRawLen("reserved0", 3)
+	d.FieldU3("warp_mode", warpModeMap)
+
+	d.FieldRawLen("unknown4", 15*8)
+	d.FieldRawLen("unknown5", 80*8)
+}
+
+func parseDolbyAtmosSupplemental(d *decode.D) {
+	d.FieldU32("dasms_sync", d.UintAssert(0xf8726fbd), scalar.UintHex)
+
+	objectCount := int64(d.FieldU16("object_count"))
+	d.FieldU8("reserved")
+
+	i := 0
+	d.FieldStructNArray("trim_configs", "trim_config", 9, func(d *decode.D) {
+		d.FieldRawLen("reserved", 7)
+		d.FieldU1("type", scalar.UintMapSymStr{
+			0: "manual",
+			1: "automatic",
+		})
+		d.FieldValueStr("config_name", trimConfigName[uint64(i)])
+
+		// TODO: this is null separted list of def strings?
+		d.FieldUTF8("raw", 14)
```

Comment:
Use same name as mediainfo? "reserved"?

Reply:
Will do for auto case, looks like there's more to do for manual, thanks again for the idea!
https://github.com/MediaArea/MediaInfoLib/blob/master/Source/MediaInfo/Audio/File_DolbyAudioMetadata.cpp#L381